src/Core/Settings.h

```diff
@@ -450,7 +450,7 @@ class IColumn;
     \
     M(Bool, compatibility_ignore_collation_in_create_table, true, "Compatibility ignore collation in create table", 0) \
     \
-    M(String, temporary_files_codec, "LZ4", "Set compression codec for temporary files (sort and join on disk). I.e. LZ4, NONE.", 0) \
+    M(String, temporary_files_codec, "LZ4", "Set compression codec for temporary files produced by (JOINs, external GROUP BY, external ORDER BY). I.e. LZ4, NONE.", 0) \
```

Comment:
We may also set the setting in https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_temporary_data_in_cache/test.py to some non-default value. Checking that proper compression is used in test may be tricky, but at lease we will see that setting do not break any functionality.

Reply:
Updated.